tests/test_log.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import pytest
+from src.log import setup_logger
+
+
+def test_logger_initialization(caplog):
+    """
+    Test that the logger initializes correctly and logs to the console.
+    """
+    with caplog.at_level(logging.INFO):
+        logger = setup_logger()
+        logger.info("This is a test log message.")
+
+    # Check that the message was logged
+    assert "This is a test log message." in caplog.text
+    assert len(caplog.records) == 1
+    assert caplog.records[0].levelname == "INFO"
+
+
+def test_logger_with_file(tmp_path):
+    """
+    Test that the logger writes to a file as expected.
+    """
+    # Create a temporary log file using tmp_path
+    log_file = tmp_path / "test_log_file.log"
+    logger = setup_logger(log_file=log_file, level=logging.INFO)
+
+    # Log a message
+    logger.info("Logging to a file!")
+
+    # Ensure the log file exists
+    assert log_file.exists()
+
+    # Read the log file and check contents
+    with log_file.open() as f:
+        logs = f.read()
+
+    assert "Logging to a file!" in logs
+
+
+@pytest.mark.parametrize("log_level", [logging.DEBUG, logging.INFO, logging.WARNING])
+def test_logger_levels(caplog, log_level):
```

Comment:
if ya wanted to be super pedantic here you could try and emit a log message for each level in each run of the test and confirm that messages from a level below the one parameterized in the test don't get emitted. also probably do the same `len(caplog.records) == 1` (or 0) to make sure you're not double-logging from the root logger